src/systemd/sd-dhcp-static-lease.h

```diff
@@ -0,0 +1,21 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <inttypes.h>
+#include <sys/types.h>
+
+#include "_sd-common.h"
+
+extern const struct hash_ops dhcp_static_leases_hash_ops;
+
+_SD_BEGIN_DECLARATIONS;
+
+typedef struct sd_dhcp_static_lease sd_dhcp_static_lease;
+typedef struct DHCPClientId DHCPClientId;
+
+int sd_dhcp_static_lease_new(DHCPClientId *client_id, const struct in_addr *address, sd_dhcp_static_lease **ret);
```

Comment:
Instead of exposing a function that takes any client of DHCPClientId (and thus come in conflict with the sd_ prefixing mentioned above), just expose a function that takes a MAC address, and make the generic function internal. The ethernet-specific one should look like this:

```c
int sd_dhcp_static_lease_new_by_mac(const uin8_t *mac_addr, size_t mac_addr_size, const struct in_addr *address, sd_dhcp_static_lease **ret);
```

I mean, I doubt many people will want anything else then match by MAC address anyway, and if people want something else we can add it later.